### ISSUE TITLE
How to implement a selectAll() function?

### ISSUE THREAD

**Issue by drmanhadan** [bignerdranch/recyclerview-multiselect#10]
I am trying to implement a selectAll() function, activate-able through the ActionBar. I was going to try to create a simple for() loop, but because setSelected requires a 'holder' object to work properly, I don't know how I would select any item that isn't being touched independently.

**Comment by MatFl** [comment]
I simply do it like this:
```
for (int i = 0; i < thumbnailAdapter.getItemCount(); i++) {
   multiSelector.setSelected(i, 0, true);
}
```

**Comment by drmanhadan** [comment]
Thanks! I had to create an ArrayList of holders, but I got the job done.
How are you able to enter the id as 0 and still have it work?

**Comment by MatFl** [comment]
If you look at the source, you can see that the ID is not used at all. Which maybe should be removed then.

https://github.com/bignerdranch/recyclerview-multiselect/blob/master/recyclerview-multiselect/src/main/java/com/bignerdranch/android/multiselector/MultiSelector.java#L87

**Comment by jingibus** [comment]
You're right — it's in the API, but is not implemented right now. That's the one thing I put in there thinking that it'd be used eventually. Long IDs show up in all the Android APIs as an optional alternative way to track adapter item identity, and I wanted to make it possible to implement that in the future.

I've never used them, though. Would love to know if other folks find them useful.

**Comment by MatFl** [comment]
I would just remove it, as it obviously confuses some users. I think the IDs are more a thing for an adapter (more or less internally). Figuring out a position for for a given ID should not be problem for most adapter implementations.
You could also have a View and Adapter which's item are not ID based at all.

**Comment by drmanhadan** [comment]
Figured this would be a good place to contribute this, mostly because I'm not good with pushing and pulling from libraries.

      public void deselectAll() {
        for (int go = mMultiSelector.getSelectedPositions().size()-1; go>=0; go--){
            mMultiSelector.setSelected(mMultiSelector
                    .getSelectedPositions().get(go), 0, false);
            }
       }

**Comment by vasiledoe** [comment]
@MatFl & @danksky  Select all and deselect all **works well** thank you!
